tests/testthat/test-merge_sce_list.R

```diff
@@ -172,6 +201,32 @@ test_that("merging SCEs with matching genes works as expected", {
     sort(assayNames(merged_sce)),
     c("counts", "logcounts")
   )
+
+  # metadata names check
+  expect_true(
+    all(c("library_id", "sample_id", "library_metadata", "sample_metadata") %in% names(metadata(merged_sce)))
+  )
+
+  # check that sample metadata is a data frame
+  expect_true(
+    is.data.frame(metadata(merged_sce)$sample_metadata)
+  )
+
+  # library metadata should contain a list of library metadata
+  expect_true(
+    length(metadata(merged_sce)$library_metadata) == 3
+  )
+
+  # check that contents of library id and sample id are correct
+  # since they all have the same library/sample id there should just be 1 rather than the list
+  expect_true(
+    metadata(merged_sce)$library_id == "library1"
+  )
```

Comment:
I think we may want to test merging with different libraries here to be sure we are doing it right for that (more common) case as well. And then you can use `expect_setequal()` to check that they are the right set of libraries.